test/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+set(SRCROOT "${PROJECT_SOURCE_DIR}/test/src")
+
+SET(SRC
+    "${SRCROOT}/Main.cpp"
+    "${SRCROOT}/UnitTests.hpp"
+    "${SRCROOT}/UnitTests.cpp"
+    "${SRCROOT}/Vector2.cpp"
+    "${SRCROOT}/Rect.cpp"
+)
+
+include_directories("${PROJECT_SOURCE_DIR}/include")
+include_directories("${PROJECT_SOURCE_DIR}/extlibs/headers")
+
+# Using sfmltest instead of test because the latter is reserved.
+add_executable(sfmltest ${SRC})
+
+target_link_libraries(sfmltest sfml-graphics sfml-window sfml-audio sfml-network sfml-system)
+
+add_custom_target(runtests ALL
```

Comment:
Using `add_test` here would be the more typical CMake way. e.g.:

```
add_test(runtests sfmltest
WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/test")
```

These will automatically get added to the "test" target, so you can easily add future test programs

Reply:
Yes, and maybe it'll get us some ctest integration?
Can't we use [this](https://github.com/catchorg/Catch2/blob/3d7104c124e3af88eda573f867f79d0aa9ef6c72/contrib/Catch.cmake#L10) for the version we're currently on?

If not, it should be on the todo list as it's pretty convenient. :)

Reply:
The linked command doesn't seem to exist or catch 1.x.

How do you run the when using `add_test`? The current solution is quite neat, as it runs the test suite when you build said target. If the same can be done with `add_test` then let me know when you figure it out, as I can't seem to get it working.

Reply:
That link's a cmake script so although it may not be in catch 1.x we could still use it in catch2 as long as it doesn't depend on any catch2 functionality.
add_test is what populates the "test" target (the one that this PR tries to avoid clashing with by naming the target "runtests". This means you can easily run the tests either by building the test target (e.g. `make test`) or just by executing `ctest` in the build directory after configuring with cmake.

there are other advantages to using the built in cmake testing stuff, for example you can add as many test programs as you want using `add_test` and they all get added to the "test" target. You can also use the built in flags for tests: `BUILD_TESTING`.

These two solutions aren't mutually exclusive either, we could easily run the proper test stuff alongside the custom command stuff if that covers all bases. 

I'll try and put together an example of this behaviour later (And also look into that catch command which supposedly improves on add_test)